lms/djangoapps/courseware/entrance_exams.py

```diff
@@ -0,0 +1,157 @@
+"""
+This file contains (or should), all entrance exam related utils/logic.
+"""
+from django.conf import settings
+
+from courseware.access import has_access
+from courseware.model_data import FieldDataCache
+from courseware.models import StudentModule
+from opaque_keys.edx.keys import UsageKey
+from student.models import EntranceExamConfiguration
+from util.milestones_helpers import get_required_content
+from util.module_utils import yield_dynamic_descriptor_descendents
+from xmodule.modulestore.django import modulestore
+
+
+def feature_is_enabled():
+    """
+    Checks to see if the Entrance Exams feature is enabled
+    Use this operation instead of checking the feature flag all over the place
+    """
+    return settings.FEATURES.get('ENTRANCE_EXAMS', False)
+
+
+def course_has_entrance_exam(course):
+    """
+    Checks to see if a course is properly configured for an entrance exam
+    """
+    if not feature_is_enabled():
+        return False
+    if not course.entrance_exam_enabled:
+        return False
+    if not course.entrance_exam_id:
```

Comment:
@dan-f -- there is currently no use case which should result in a valid-but-falsy entrance_exam_id value -- basically a course will either have a value or it will be set to None.  Does that help to clarify?